src/gadget.hpp

```diff
@@ -46,16 +46,22 @@ class l_gadget : public gadget<FieldT> {
     pb_variable_array<FieldT> intermediate_val1;
     pb_variable_array<FieldT> intermediate_val2;
     pb_variable_array<FieldT> intermediate_val3;
+    pb_variable_array<FieldT> intermediate_val4;
+    pb_variable_array<FieldT> intermediate_val5;
 
     std::shared_ptr<multipacking_gadget<FieldT> > unpack_inputs; /* multipacking gadget */
 
     std::shared_ptr<digest_variable<FieldT>> h1_var; /* H(R1) */
     std::shared_ptr<digest_variable<FieldT>> h2_var; /* H(R2) */
     std::shared_ptr<digest_variable<FieldT>> h3_var; /* H(R2) */
+    std::shared_ptr<digest_variable<FieldT>> h4_var; /* H(R2) */
+    std::shared_ptr<digest_variable<FieldT>> h5_var; /* H(R2) */
 
     std::shared_ptr<digest_variable<FieldT>> r1_var; /* R1 */
     std::shared_ptr<digest_variable<FieldT>> r2_var; /* R2 */
     std::shared_ptr<digest_variable<FieldT>> r3_var; /* R3 */
+    std::shared_ptr<digest_variable<FieldT>> r4_var; /* R3 */
+    std::shared_ptr<digest_variable<FieldT>> r5_var; /* R3 */
```

Comment:
The comments might be confusing rather than helpfull, `r4_var` shouldn't be `R3` etc

Reply:
fixed...